modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/SelectionFilter.js

```diff
@@ -55,50 +108,152 @@ const getOdataString = ({entityFieldType, id, selectedData}) => {
 	}
 
 	const quotedItemValues = itemsValues.map((itemValue) => {
-		return typeof itemValue === 'string' ? `'${itemValue}'` : itemValue;
+		const value =
+			typeof itemValue === 'object' ? itemValue.value : itemValue;
+
+		return typeof value === 'string' ? `'${value}'` : value;
```

Comment:
That would be what you mentioned in a different comment, migrating use cases that still pass `itemValues` as string arrays instead of object arrays with all the info of the item. Once all are migrated to object arrays, the value should always be in `itemValue.value`. I've removed the `object` check and we'll see what breaks in the CI.